experiments/Standalone/Soil/water_conservation.jl

```diff
@@ -0,0 +1,262 @@
+    CTS.NewtonsMethod(
+        max_iters = 50,
+        update_j = CTS.UpdateEvery(CTS.NewNewtonIteration),
+        convergence_checker = conv_checker,
+    ),
+)
+
+t_start = FT(0)
+t_end = FT(1e6)
+dts = FT.([10, 100, 1000, 10000])
+
+# van Genuchten parameters for clay (from Bonan 2019 supplemental program 8.2)
+ν = FT(0.495)
+K_sat = FT(0.0443 / 3600 / 100) # m/s
+vg_n = FT(1.43)
+vg_α = FT(0.026 * 100) # inverse meters
+θ_r = FT(0.124)
+S_s = FT(1e-3) #inverse meters
+hcm = vanGenuchten(; α = vg_α, n = vg_n)
+
+zmax = FT(0)
+zmin = FT(-0.5)
+nelems = 50
+
+params = Soil.RichardsParameters{FT, typeof(hcm)}(ν, hcm, K_sat, S_s, θ_r)
+soil_domain = Column(; zlim = (zmin, zmax), nelements = nelems)
+sources = ()
+
+# Set flux boundary conditions (used for calculating mass balance)
+flux_in = FT(-1e-7)
+top_bc = Soil.FluxBC((p, t) -> eltype(t)(flux_in))
+flux_out = FT(0)
+bot_bc = Soil.FluxBC((p, t) -> eltype(t)(flux_out))
+
+boundary_fluxes = (; top = (water = top_bc,), bottom = (water = bot_bc,))
+
+soil = Soil.RichardsModel{FT}(;
+    parameters = params,
+    domain = soil_domain,
+    boundary_conditions = boundary_fluxes,
+    sources = sources,
+)
+
+exp_tendency! = make_exp_tendency(soil)
+imp_tendency! = make_imp_tendency(soil)
+update_jacobian! = make_update_jacobian(soil)
+
+rmses = Array{FT}(undef, length(dts))
+mass_errors = Array{FT}(undef, length(dts))
+for i in eachindex(dts)
+    dt = dts[i]
+
+    Y, p, coords = initialize(soil)
+    @. Y.soil.ϑ_l = FT(0.24)
+
+    jac_kwargs =
+        (; jac_prototype = RichardsTridiagonalW(Y), Wfact = update_jacobian!)
+
+    prob = ODE.ODEProblem(
+        CTS.ClimaODEFunction(
+            T_exp! = exp_tendency!,
+            T_imp! = ODE.ODEFunction(imp_tendency!; jac_kwargs...),
+            dss! = ClimaLSM.dss!,
+        ),
+        Y,
+        (t_start, t_end),
+        p,
+    )
+
+    sol = ODE.solve(prob, ode_algo; dt = dt, saveat = dt)
+
+    # Calculate water mass balance over entire simulation
+    mass_end = sum(sol.u[end].soil.ϑ_l)
+    mass_start = sum(sol.u[1].soil.ϑ_l)
+    t_sim = sol.t[end] - sol.t[1]
+    # Flux changes water content every timestep (assumes constant flux_in, flux_out)
+    mass_change_exp = -(flux_in - flux_out) * t_sim
+    mass_change_actual = mass_end - mass_start
+    relerr = abs(mass_change_actual - mass_change_exp) / mass_change_exp
+
+    @assert relerr < FT(1e-9)
+    mass_errors[i] = relerr
+
+    # Compute RMSE vs reference solution (found using dt = 1s)
+    rmses[i] = rmse(ref_soln_flux, parent(sol.u[end].soil.ϑ_l))
+end
+
+# Save flux BC mass conservation error and RMSE as artifact
+plt = Plots.plot()
+plt_twin = twinx(plt)
+Plots.plot!(
+    plt,
+    dts,
+    rmses,
+    label = "RMSE",
+    color = "red",
+    linewidth = 3,
+    xlabel = "Δt (s)",
+    ylabel = "RMSE ϑ",
+    background_color_legend = nothing,
+    xaxis = :log10,
+    yaxis = :log10,
+    xticks = dts,
+    title = "RMSE and Water Conservation with Flux BCs",
```

Comment:
I tried moving the legend and using `annotate!`, but neither of them looked very good. I decided to move the legends to the bottom of the plot, which I think looks at least a little bit nicer because the labels are closer to the lines they refer to.

Reply:
ok! we can figure it out later when we need to make the publication quality. looks great otherwise!